fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -91,18 +99,73 @@ public void analyzeIgnoreSlot(Expr expression, AnalyzeState analyzeState, Scope
         bottomUpAnalyze(visitor, expression, scope);
     }
 
-    private void bottomUpAnalyze(Visitor visitor, Expr expression, Scope scope) {
-        for (Expr expr : expression.getChildren()) {
+    private boolean isHighOrderFunction(Expr expr) {
+        if (expr instanceof FunctionCallExpr) {
+            // expand this in the future.
+            if (((FunctionCallExpr) expr).getFnName().getFunction().equalsIgnoreCase(FunctionSet.ARRAY_MAP)) {
```

Comment:
All function names are lowercase.

Reply:
`equalsIgnoreCase ` is also used in many places.

Reply:
we will remove old usage, new codes shouldn't use equalsIgnoreCase

Reply:
ok